examples/workflows/script-artifact-passing.yaml

```diff
@@ -9,11 +9,7 @@ spec:
     steps:
     - - name: generate-artifact
         template: whalesay
-    - - arguments:
```

Comment:
TODO: find out why this is broken

Reply:
I figure out why it's broken. PR up at #648

Leaving this here as notes for my debugging process:

`_get_arguments` in CallableTemplateMixin isn't finding your assignment of `message`

E.g. we know this works:
```py
        print_message(name="consume-artifact", arguments=ga.get_artifact("hello-art").as_name("message"))
```

But when we try
```py
        print_message(name="consume-artifact", message=ga.get_artifact("hello-art"))
```

This is assigning the `message` kwarg to an artifact that looks like
```
Artifact(name='hello-art', from_='{{steps.generate-artifact.outputs.artifacts.hello-art}}', path='/tmp/hello_world.txt', ...)
```

So _get_arguments seeing the given artifact as it's hiding in the `message` kwarg, plus it should have the name "message" otherwise the step will fail. The fix is to look through the kwargs for anything that looks like a Parameter or Artifact, and rename it to the kwarg name. Bonus is that we can check for clashing kwargs too!